nginx-controller/nginx/configurator.go

```diff
@@ -84,6 +84,16 @@ func (cnf *Configurator) addOrUpdateMergableIngress(mergeableIngs *MergeableIngr
 	var upstreams []Upstream
 	var keepalive string
 
+	for _, blacklistAnn := range masterBlacklist {
+		for _, masterAnn := range mergeableIngs.Master.Ingress.Annotations {
```

Comment:
would it be more efficient if we iterate over all master annotations and look up if a particular annotation is in the black list using a map?

Reply:
I optimized it, since annotations are already in a map[string]string, I just look over all the blacklist items and if any are in the master they are removed. Looking through the blacklist since its small and there could be many annotations in an Ingress Resource.